raiden/blockchain_events_handler.py

```diff
@@ -364,18 +357,69 @@ def handle_channel_batch_unlock(raiden: 'RaidenService', event: Event):
     block_number = data['block_number']
     block_hash = data['block_hash']
     transaction_hash = data['transaction_hash']
+    participant1 = args['participant']
+    participant2 = args['partner']
+    locksroot = args['locksroot']
 
     chain_state = views.state_from_raiden(raiden)
+    token_network_state = views.get_token_network_by_identifier(
+        chain_state,
+        token_network_identifier,
+    )
+    assert token_network_state is not None
+
+    if participant1 == raiden.address:
+        partner = participant2
+    elif participant2 == raiden.address:
+        partner = participant1
+    else:
+        log.debug(
+            "Discarding unlock event, we're not part of it",
+            participant1=pex(participant1),
+            participant2=pex(participant2),
+        )
+        return
+
+    channel_identifiers = token_network_state.partneraddresses_to_channelidentifiers[partner]
+    canonical_identifier = None
+
+    for channel_identifier in channel_identifiers:
+        state_change_record = get_state_change_with_balance_proof_by_locksroot(
```

Comment:
We can figure out which end of the channel was unlocked by checking the value of `participant`, and then only one of the two queries are necessary (this will reduce the queries by half). Additionally, maybe it makese sense to query with a `IN` clause? and let the database do the work of looping over the entries.